app/Http/Services/PurchaseOrderService.php

```
@@ -128,4 +129,21 @@ private static function generatePoNumber()
         $newNumber = str_pad($lastRefNo + 1, 6, '0', STR_PAD_LEFT);
         return "PO-{$year}-{$newNumber}";
     }
+    public static function setServed(RequestPurchaseOrder $requestPurchaseOrder)
+    {
+        $mrrItemsQuery = $requestPurchaseOrder->mrrNcpoItems();
+        $allItemsAccepted = $mrrItemsQuery->get()->every(function ($item) {
+            return $item->acceptance_status === ReceivingAcceptanceStatus::ACCEPTED->value;
+        });
+        if (!$allItemsAccepted) {
+            throw new \Exception('Cannot mark Purchase Order as SERVED. Some items are still pending acceptance.');
```

PHPMD (2.15.0)

Missing class import via use statement (line '139', column '23'). (undefined)

(MissingImport)